test/common/alarm/AlarmFactory.test.ts

```diff
@@ -409,3 +410,32 @@ test("addAlarm: disambigatorAction takes precedence over default action", () =>
 
   expect(alarm.action).toStrictEqual(snsAction);
 });
+
+test("addAlarm: custom alarm naming strategy", () => {
+  const alarmName = "alarmName";
+  const alarmLabel = "alarmLabel";
+  const alarmDedupe = "alarmDedupe";
+  const stack = new Stack();
+  const customNamingStrategy: IAlarmNamingStrategy = {
+    // @ts-ignore
+    getName: (props: AlarmNamingInput) => alarmName, 
+    // @ts-ignore
+    getWidgetLabel: (props: AlarmNamingInput) => alarmLabel, 
+    // @ts-ignore
+    getDedupeString: (props: AlarmNamingInput) => alarmDedupe, 
```

Comment:
I do get some complaints when running locally that it doesn't like the fact that props is never used, but it is necessary when implementing the interface.

This may be an issue, but curious on your thoughts

Reply:
I'll fix this in another Pull Request. I have a better solution.